File: src/main/scala/org/xbery/artbeams/common/antispam/domain/AntispamQuiz.scala
package org.xbery.artbeams.common.antispam.domain

/**
  * Antispam quiz.
  *
  * @author Radek Beran
  */
@SerialVersionUID(-1L)
case class AntispamQuiz(
  question: String,
  answer: String,
) extends Serializable

object AntispamQuiz {
  final val CacheName = "antispam_quizes"

  lazy val Empty = AntispamQuiz(
    "",
    ""
  )
}

File: src/main/kotlin/org/xbery/artbeams/common/antispam/domain/AntispamQuiz.kt
package org.xbery.artbeams.common.antispam.domain

import org.xbery.artbeams.common.repository.IdentifiedEntity
import java.io.Serializable

/**
 * Antispam quiz.
 *
 * @author Radek Beran
 */
data class AntispamQuiz(
    val question: String,
    val answer: String
) : IdentifiedEntity, Serializable {

    override val id: String
        get() = question

    companion object {
        const val CacheName = "antispam_quizes"

        val Empty = AntispamQuiz(
                "",
                ""
        )
    }
}
